client/my-sites/plugins/plugins-list/index.tsx

```diff
@@ -30,48 +29,86 @@ import { isJetpackSite } from 'calypso/state/sites/selectors';
 import { getSelectedSite, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 import PluginManagementV2 from '../plugin-management-v2';
 import { getPluginActionDailogMessage, getSitePlugin, handleUpdatePlugins } from '../utils';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { Plugin, PluginFilter, PluginStatus } from 'calypso/state/plugins/installed/types';
 
 import './style.scss';
 
-function checkPropsChange( nextProps, propArr ) {
-	let i;
-	let prop;
-
-	for ( i = 0; i < propArr.length; i++ ) {
-		prop = propArr[ i ];
-
-		if ( ! isEqual( nextProps[ prop ], this.props[ prop ] ) ) {
-			return true;
-		}
-	}
-	return false;
+interface Props {
+	hasManagePlugins?: boolean;
+	header: string;
+	inProgressStatuses: PluginStatus[];
+	isPlaceholder: boolean;
+	plugins: Plugin[];
+	selectedSite?: SiteDetails;
+	selectedSiteSlug?: string;
+	siteIsAtomic?: boolean;
+	siteIsJetpack?: boolean;
+	searchTerm: string;
+	recordGoogleEvent: typeof recordGoogleEvent;
+	allSites: ReturnType< typeof getSites >;
+	pluginsOnSites: ReturnType< typeof getPluginsOnSites >;
+	removePluginStatuses: typeof removePluginStatuses;
+	translate: typeof translate;
```

Comment:
As mentioned above, we should be able to use the `translate` type from `i18n-calypso` directly.